### PR TITLE
Add more details for note command in DG

### PR DESCRIPTION
**Pull request by wang-jun-hao**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -149,6 +149,28 @@ The following sequence diagram shows how note adding operation works:
 
 ![NoteSequenceDiagramSD](images/NoteSequenceDiagramSDUpdatePatientInModel.png)
 
+Step 1. The user launches the application and `find` the patient to access. 
+
+Note: Every `patient` in the `model` has a `MedicalNoteList` that is initialised as an empty list at 
+program start-up to optimise start-up time.
+
+Step 2. The user then `access` the patient using the index of the patient in the filtered list.
```
Comment (foojingyi) [inline docs/DeveloperGuide.md]: Sounds grammatically incorrect to say "the user then _access_" even though it is clear in the markup that this is a command. Is there a way to rephrase this sentence?

**Review by foojingyi — COMMENTED** [review]
Just one comment, otherwise LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (wang-jun-hao) [reply docs/DeveloperGuide.md]: fixed

**Review by foojingyi — APPROVED** [review]
LGTM!

**Pull request merged**